### ISSUE TITLE
Error if text > 200 char

### ISSUE THREAD

**Issue by antoniofagardo** [pndurette/gTTS#252]
I am currently running version `2.2.0` and it seems like if the text we are trying to convert is greater than 200 char, the conversion fails with the following message:

```
Traceback (most recent call last):
  File "main.py", line 11, in <module>
    myobj.save("audiobook.mp3")
  File "/usr/local/lib/python3.8/dist-packages/gtts/tts.py", line 311, in save
    self.write_to_fp(f)
  File "/usr/local/lib/python3.8/dist-packages/gtts/tts.py", line 293, in write_to_fp
    raise gTTSError(tts=self, response=r)
gtts.tts.gTTSError: 200 (OK) from TTS API. Probable cause: Unknown
```

Sample code to reproduce the problem:
```
from gtts import gTTS

finalChar = 201
language = 'en'
finalText = ""
for i in range(30):
    finalText += "\n0123456789"
print(finalText[:finalChar])

myobj = gTTS(text=finalText[:finalChar], lang=language, slow=False)
myobj.save("audiobook.mp3")
```

if you change back `finalChar` to `200`, the request works and the file is written on disk.

**Comment by pndurette** [comment]
@antoniofagardo Thanks for the report, let me take a look.

**Comment by pndurette** [comment]
Yeah, there's a weird bug with how requests are prepared I think. It sends way too much data to the API. Investigating.

**Comment by pndurette** [comment]
Ah found it—
`_package_rpc()` takes the entire text submitted and not a part. So say it splits the text in 10 parts, the entire text is sent 10 times.

**Comment by pndurette** [comment]
Pushes `gTTS` `2.2.1` fixing the problem! Thanks for the bug report!

**Comment by antoniofagardo** [comment]
Thank you for the quick fix!

**Comment by KelvinKramp** [comment]
Ive got the same problem, but i think due to a different reason

Traceback (most recent call last):
  File "/Users/mackliniek/Tests/morfine_protocol.py", line 32, in <module>
    welcome()
  File "/Users/mackliniek/Tests/morfine_protocol.py", line 30, in welcome
    SR.text_speech(morfine_conversation[0])
  File "/Users/mackliniek/Tests/TTS_SST.py", line 28, in text_speech
    speech.save('speech.mp3')
  File "/Users/mackliniek/dashboard/venv/lib/python3.9/site-packages/gtts/tts.py", line 312, in save
    self.write_to_fp(f)
  File "/Users/mackliniek/dashboard/venv/lib/python3.9/site-packages/gtts/tts.py", line 294, in write_to_fp
    raise gTTSError(tts=self, response=r)
gtts.tts.gTTSError: 200 (OK) from TTS API. Probable cause: Unknown

Is there any fix? I have installed gTTS 2.2.1.

**Comment by AngelosDouglas** [comment]
Mesmo problema aqui

**Comment by KelvinKramp** [comment]
I found the problem in my case, the translation accent was set to "en-ca" representing english canadian. After some debugging (sweat over my body,  tears dropping down my head, axillary region all wet etc. :-)) I found it. Maybe you should check what happens if you change the speaking accent to English with "en", I also removed slow=False. Check it out and let me know if it works out.

**Comment by Lucas-Uchiha** [comment]
In my tests 'pt-br' and 'pt-pt' are not working, but only 'pt' works. Any idea why this is happening?

**Comment by srugai** [comment]
Unbelievable, the code below was working like a charm for two months now (gTTs 2.2.1). Started to get the error commented here. Tried KelvinKramp approach and it is working again. I mean the error stopped but lang="en" and lang="en-au" produces a different result right? 

def text_to_speech():
    tts = gtts.gTTS(T.get("1.0", tk.END), lang='en-au') # 'en-au': 'English (Australia)'
    mp3_file = "_temp.mp3"
    tts.save(mp3_file)
    music = AudioSegment.from_mp3(mp3_file)
    print("Playing mp3 file...")
    play(music)

**Comment by pndurette** [comment]
@KelvinKramp and @srugai 

Hi there, you're right, Google has removed upstream almost of the `<Lang>-<geo>` language coded that used to work (and yeah produce different accents). 

I've removed them, see #267, but since they've been part of the library for years, I'm adding a compatibility fallback (eg "en-CA" falls back to "en").

**Comment by NikithaNimbalkar** [comment]
> @KelvinKramp and @srugai
> 
> Hi there, you're right, Google has removed upstream almost of the `<Lang>-<geo>` language coded that used to work (and yeah produce different accents).
> 
> I've removed them, see #267, but since they've been part of the library for years, I'm adding a compatibility fallback (eg "en-CA" falls back to "en").

What to do if we actually want the accents...

**Comment by evinracher** [comment]
I was having the same issue. My code was:

`from gtts import gTTS
tts = gTTS('Hola mundo. Estamos convirtiendo texto a voz con Python.',
           lang='es-us')`

I chante es-us to es and it worked

**Comment by erichsu0527** [comment]
Since zh is not a language, Chinese does not work at all. 

`
from gtts import gTTS
tts = gTTS("说中文", lang="zh-CN")
tts.save("helloch.mp3")

tts = gTTS("說中文", lang="zh-TW")
tts.save("helloch.mp3")
`

Get "gtts.tts.gTTSError: 200 (OK) from TTS API. Probable cause: Unknown" on both tts.save()

**Comment by zesto01** [comment]
I am having the same issue with erichsu0527. Are there any workarounds?

**Comment by joeminicucci** [comment]
Same problem here as @zesto01 and @erichsu0527.
@pndurette I was having trouble finding the root cause debugging the code, would you be able to take a look? Would much appreciate your trouble as its effecting many users for an open source downstream I maintain.

### Input:
```
tts = gTTS('的', lang='zh-CN', lang_check=True)
tts.save(self.path)
```

### Trace:
```
  File "C:\Users\User1\AppData\Roaming\Anki2\addons21\chinese\tts.py", line 50, in download
    self.func()
  File "C:\Users\User1\AppData\Roaming\Anki2\addons21\chinese\tts.py", line 56, in get_google
    tts.save(self.path)
  File "C:\Users\User1\AppData\Roaming\Anki2\addons21\venv\lib\site-packages\gtts\tts.py", line 312, in save
    self.write_to_fp(f)
  File "C:\Users\User1\AppData\Roaming\Anki2\addons21\venv\lib\site-packages\gtts\tts.py", line 294, in write_to_fp
    raise gTTSError(tts=self, response=r)

gtts.tts.gTTSError: 200 (OK) from TTS API. Probable cause: Unknown
```

**Comment by pndurette** [comment]
@erichsu0527, @zesto01, @joeminicucci

I was able to run all of your code samples without any issues. Have you updated? There was an upstream with languages, which was fixed in `2.2.2`

**Comment by pndurette** [comment]
@KelvinKramp, @srugai & NikithaNimbalkar:

The accents are back, just obtained differently!

I've added updated documentation on how to obtain local _accents_ with the combination of the right language code (`lang`) and top-level domain (`tld`)! See: https://gtts.readthedocs.io/en/latest/module.html#localized-accents

**Comment by joeminicucci** [comment]
@pndurette I tested that code 5 days ago on 2.2.2. Let me try again.

**Comment by joeminicucci** [comment]
@pndurette Following the new tld convention (doesn't apply since Im using zh-CN and zh-TW only), Im still reliably reproducing the error.

**Comment by pndurette** [comment]
> @pndurette Following the new tld convention (doesn't apply since Im using zh-CN and zh-TW only), Im still reliably reproducing the error.

@joeminicucci Ah damn, let's look. Do you mind opening a new issue with your problem? It's really getting difficult to follow exactly the issues everyone is experiencing as this issue has went all over the place.

**Comment by zesto01** [comment]
@joeminicucci @pndurette 
I tried using 2.2.2 and lang="zh-CN", it is now working. though parameter slow=True, does no longer work.

**Comment by agnplu** [comment]
@pndurette 
Will other local accents be available too? I only see zh-CN and zh-TW now, and I'm mostly interested in en-US.

**Comment by pndurette** [comment]
@agnplu Please open new a Discussion for questions—but for "en-US", see [here](https://gtts.readthedocs.io/en/latest/module.html#localized-accents). You should be able to use `lang='en'` with `tld='com'` (which is the default)
